### DISCUSSION TITLE
Static types

### DISCUSSION THREAD

**Discussion by andrecasal** [joe-bell/cva#213] [Q&A] [answered]
I see the `cva()` function takes a generic type.

Depending on how this is done, it might be useful to have something like:
```tsx
type HeadingVariantProps = {
  weight?: 'thin' | 'extralight' | 'light' | 'normal' | 'medium' | 'semibold' | 'bold' | 'extrabold' | 'black'
  align?: 'left' | 'center' | 'right'
}

const headingVariants = cva<HeadingVariantProps>({...})
```

**Is this possible, and advisable? Can you please provide an example?**

**Comment by andrecasal** [comment]
This would also simplify type overview:
![Screenshot 2023-09-08 at 16 45 35](https://github.com/joe-bell/cva/assets/22457463/c8e23300-4adc-4271-804a-2f75a8773f3e)

**Comment by rjfranc0** [comment]
Completely agree, is this comming ?

**Comment by joe-bell** [comment, marked as answer]
It's not something I'm planning I'm afraid, my intent is to make `cva` the single source of truth for types; not something you can customise
